### PR TITLE
Improve UFS journal flush/replay failure handling

### PR DESCRIPTION
**Pull request by ggezer**
For v1 UFS journals, flush failure handling will leave journal system in a state where 
- current log file is uncompleted. 
- subsequent journal writes will cause a new log file creation.
Since secondary journal skips incomplete files, this will create a gap in journal sequences it sees and its background thread will crash.

This PR:
- fixes flush failure handling. 
- crashes secondary process crashes when journal replaying encounters an issue.

### CODE REVIEW

**Comment by ggezer** [comment]
@calvinjia PTAL. Thanks!

**Comment by calvinjia** [comment]
@ggezer Is there an easy way to test this?

**Comment by ggezer** [comment]
> @ggezer Is there an easy way to test this?

@calvinjia Added a unittest for new journal writer behaviour. 
The whole scenario would be complicated to set up.

File: core/server/master/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java [hunk 1]
```diff
@@ -367,6 +399,7 @@ private DataOutputStream createMockDataOutputStream(UfsJournalLogWriter writer)
       throws IOException {
     flushOutputStream(writer);
     DataOutputStream badOut = Mockito.mock(DataOutputStream.class);
+    Whitebox.setInternalState(badOut, "written", 1);
```
Comment (calvinjia) [inline core/server/master/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java]: Can you update the javadoc for this method indicating that the stream is not wellformed?

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Comment by ggezer** [comment]
@calvinjia Should we backport this to 1.8?

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-1.8 please.

**Comment by calvinjia** [comment]
@ggezer unfortunately it looks like you will need to cherry-pick it manually to 1.8

**Pull request merged**